packages/bot/src/stores/quick-strategy-store.js

```diff
@@ -98,41 +110,128 @@ export default class QuickStrategyStore {
             this.market_dropdown = market_options;
         });
     }
 
     @action.bound
-    async updateTradetypeDropdown(symbol = this.initial_values.symbol) {
+    // eslint-disable-next-line
+    async onChangeMarketDropdown(setFieldValue, value) {
+        setFieldValue('symbol', value);
+        await this.updateTradetypeDropdown(setFieldValue, value);
+
+        const first_trade_type_option = this.trade_type_dropdown[Object.keys(this.trade_type_dropdown)[0]][0].value;
+
+        setFieldValue('trade_type', first_trade_type_option);
+    }
+
+    @action.bound
+    async updateTradetypeDropdown(setFieldValue, symbol = this.initial_values.symbol) {
         const { contracts_for }  = ApiHelper.instance;
         const trade_type_options = await contracts_for.getGroupedTradeTypes(symbol);
+        const first_trade_type_option = trade_type_options[Object.keys(trade_type_options)[0]][0].value;
 
-        runInAction(() => {
-            this.trade_type_dropdown = trade_type_options;
-        });
+        runInAction(() => this.trade_type_dropdown = trade_type_options);
+
+        if (setFieldValue) {
+            setFieldValue('trade_type', first_trade_type_option);
+    
+            await this.updateDurationDropdown(setFieldValue, symbol, first_trade_type_option);
+        }
     }
 
     @action.bound
     // eslint-disable-next-line
-    async onChangeMarketDropdown(setFieldValue, value) {
-        setFieldValue('symbol', value);
-        await this.updateTradetypeDropdown(value);
+    async onChangeTradeTypeDropdown(setFieldValue, symbol, trade_type) {
+        setFieldValue('trade_type', trade_type);
 
-        const first_option = this.trade_type_dropdown[Object.keys(this.trade_type_dropdown)[0]][0].value;
-        setFieldValue('tradetype', first_option);
+        await this.updateDurationDropdown(setFieldValue, symbol, trade_type);
+    }
+
+    @action.bound
+    async updateDurationDropdown(
+        setFieldValue,
+        symbol = this.initial_values.symbol,
+        trade_type = this.initial_values.trade_type
+    ) {
+        const { contracts_for } = ApiHelper.instance;
+        const durations = await contracts_for.getDurations(symbol, trade_type);
+        const first_duration_option = durations[Object.keys(durations)[0]].unit;
+
+        runInAction(() => this.duration_dropdown = durations);
+
+        if (setFieldValue) {
+            setFieldValue('duration_type', first_duration_option);
+    
+            this.updateDurationValue(setFieldValue, first_duration_option);
+        }
     }
 
     @action.bound
     // eslint-disable-next-line
-    onChangeTradeTypeDropdown(setFieldValue, value) {
-        setFieldValue('tradetype', value);
+    async onChangeDurationDropdown(setFieldValue, duration_type) {
+        setFieldValue('duration_type', duration_type);
+
+        this.updateDurationValue(setFieldValue, duration_type);
+    }
+
+    @action.bound
+    async updateDurationValue(
+        setFieldValue,
```

Comment:
Same here, since `setFieldValue` does not always have a value it would be good to mark this as an optional argument by supplying a default value of e.g. `null`.